src/linear_algebra/span.lean

```diff
@@ -557,6 +557,22 @@ begin
   exact ⟨y, ⟨hyd, by simpa only [span_le, singleton_subset_iff]⟩⟩,
 end
 
+/-- The span of a finite subset is compact in the lattice of submodules. -/
+lemma finite_span_is_compact_element (S : finset M) :
+  complete_lattice.is_compact_element (span R S : submodule R M) :=
+begin
+  rw span_eq_supr_of_singleton_spans,
+  simp only [finset.mem_coe],
+  rw ←finset.sup_eq_supr,
+  exact complete_lattice.finset_sup_compact_of_compact S
+    (λ x _, singleton_span_is_compact_element x),
+end
+
+/-- The span of a finite subset is compact in the lattice of submodules. -/
+lemma finite_span_is_compact_element' (S : set M) (h : S.finite) :
+  complete_lattice.is_compact_element (span R S : submodule R M) :=
+finite.coe_to_finset h ▸ (finite_span_is_compact_element h.to_finset)
```

Comment:
```suggestion
lemma finite_span_is_compact_element (S : set M) (h : S.finite) :
  complete_lattice.is_compact_element (span R S : submodule R M) :=
finite.coe_to_finset h ▸ (finset_span_is_compact_element h.to_finset)
```

Reply:
Or maybe
```suggestion
lemma is_compact_element_span_of_finite (S : set M) (h : S.finite) :
  complete_lattice.is_compact_element (span R S : submodule R M) :=
finite.coe_to_finset h ▸ (is_compact_element_span_finset h.to_finset)
```
Where the `of` is recommended by the naming convention